server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateService.java

```diff
@@ -1303,7 +1306,7 @@ public static List<CompressedXContent> collectMappings(
             if (template.getDataStreamTemplate().isAllowCustomRouting()) {
                 mappings.add(0, DEFAULT_TIMESTAMP_MAPPING_WITH_ROUTING);
             } else {
-                mappings.add(0, DEFAULT_TIMESTAMP_MAPPING);
+                mappings.add(0, DEFAULT_TIMESTAMP_MAPPING_WITHOUT_ROUTING);
```

Comment:
Note for reviewer: note that I had to use `"_routing": {"required": false}` explicitly here, instead of relying on the default, because otherwise there is no way to detect contradiction between mapping and settings coming from `ComposableIndexTemplate.DataStreamTemplate`, as they both get translated into a single `RoutingFieldMapper`.
Since this code is executed only if `template.getDataStreamTemplate() != null`, it should be same